apps/src/code-studio/components/aiTutor/compilationTutor.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {useSelector, useDispatch} from 'react-redux';
+import Button from '@cdo/apps/templates/Button';
+import style from './ai-tutor.module.scss';
+import {askAITutor} from '@cdo/apps/aiTutor/redux/aiTutorRedux';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+
+// AI Tutor feature that explains to students why their code did not compile.
+const CompilationTutor = ({levelId}) => {
+  const dispatch = useDispatch();
+  const javalabState = useSelector(state => state.javalab);
```

Comment:
Instead of getting all the state here I think we should get individual pieces as separate constants (for example, `const hasRunOrTestedCode = useSelector(state => state.javalab.hasRunOrTestedCode);`. This will prevent extra re-renders when something changes in the java lab state that we don't care about in this component.